LamestWebserver/LamestWebserver/Core/Logger.cs

```diff
@@ -261,26 +315,91 @@ public void CrashAndBurn(string msg)
 
         internal void Log(ELoggingLevel loggingLevel, string msg)
         {
-            // TODO: Implement propper logging!
-
-            ServerHandler.LogMessage($"[{loggingLevel.ToString()} \\\\\\\\ {msg}]");
+            if (currentOutputSource != EOutputSource.None)
+            {
+                streamWriter.WriteLine($"[{loggingLevel.ToString()} \\\\\\\\ {msg}]");
+            }
         }
 
         /// <summary>
         /// Closes and Flushes the Logger stream.
         /// </summary>
         public void Close()
         {
-            // TODO: Close and Flush stream!
-            DebugExcept(new NotImplementedException());
+                streamWriter.Flush();
+                streamWriter.Close();
         }
 
         /// <summary>
         /// Opens and initializes the steam to write to.
         /// </summary>
         protected void Open()
         {
-            DebugExcept(new NotImplementedException());
+
+            ConfigureMultiStream();
+
+            if (currentOutputSource != EOutputSource.None)
+                streamWriter = new StreamWriter(multiStream);
+
+        }
+
+        internal void ConfigureMultiStream()
+        {
+
+            multiStream.Streams.Clear();
+            switch (currentOutputSource)
+            {
+                case EOutputSource.None:
```

Comment:
to be honest, this is not a future-proof solution. I'd prefer walking over the EOutputSource enum and checking
```C#
if((currentOutputSource & EEnumEntry) != EEnumEntry )
    AddStream(Console.OpenStandardOutput());
```

Reply:
👍